### PR TITLE
refactor(components): [el-pagination] optimize component style

### PR DESCRIPTION
**Pull request by msidolphin**
Please make sure these boxes are checked before submitting your PR, thank you!

- [x] Make sure you follow contributing guide [English](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.en-US.md) | ([中文](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.zh-CN.md) | [Español](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.es.md) | [Français](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.fr-FR.md)).
- [x] Make sure you are merging your commits to `dev` branch.
- [x] Add some descriptions and refer to relative issues for your PR.

### CODE REVIEW

File: packages/components/pagination/src/components/sizes.vue [hunk 1]
```diff
@@ -41,6 +41,10 @@ const paginationSizesProps = buildProps({
     default: '',
   },
   disabled: Boolean,
+  size: {
+    type: String,
+    default: 'default',
+  },
```
Comment (JeremyWuuuuu) [inline packages/components/pagination/src/components/sizes.vue:47]: Can we use the prop defined in `useSizes`?

File: packages/components/pagination/src/pagination.ts [hunk 2]
```diff
@@ -234,6 +234,15 @@ export default defineComponent({
...
     }
 
+    function addClass(element: any, cls: string) {
+      if (element) {
+        if (!element.props) {
+          element.props = {}
+        }
+        element.props.class = [element.props.class, cls].join(' ')
+      }
+    }
```
Comment (JeremyWuuuuu) [inline packages/components/pagination/src/pagination.ts:244]: I don't think changing the props here is a good idea, can we use `provide/inject` to do the same thing here?

Comment (JeremyWuuuuu) [reply packages/components/pagination/src/pagination.ts:244]: Can we do something else here? Exposing API to the parent for mutating the state.

```ts
element.someMethod()
```

**Review by jw-foss — APPROVED** [review]
I left some minor comments for you to consider, let me know if you have any questions ;>

File: packages/components/pagination/src/pagination.ts (hunk 2 above)
Comment (msidolphin) [reply packages/components/pagination/src/pagination.ts:244]: This method is modifies the class of the vnode, so I don't think it's necessary to exposing API

File: packages/components/pagination/src/components/sizes.vue (hunk 1 above)
Comment (msidolphin) [reply packages/components/pagination/src/components/sizes.vue:47]: No, the size only have default and small

**Comment by JeremyWuuuuu** [comment]
Let's roll it

**Pull request merged**